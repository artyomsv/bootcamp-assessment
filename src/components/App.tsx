import * as React from 'react';
import {createStyles, Theme, withStyles, WithStyles} from '@material-ui/core';
import Header from './header';
import NavigationDrawer from './navigation/drawer';
import ActorsView from './views/actors';
import actors from './views/actors.mock.json';

const styles = (theme: Theme) => createStyles({
  root: {
    flexGrow: 1,
    // height: 430,
    zIndex: 1,
    overflow: 'hidden',
    position: 'relative',
    display: 'flex',
  },
});

class App extends React.Component<WithStyles<typeof styles>> {

  constructor(props: WithStyles<typeof styles>) {
    super(props);
  }

  render() {
    const {classes} = this.props;
    return (
      <div className={classes.root}>
        <Header/>
        <NavigationDrawer/>
        <ActorsView actors={actors}/>
      </div>
    );
  }
}

export default withStyles(styles)(App);
