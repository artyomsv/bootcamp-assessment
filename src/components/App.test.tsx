import * as React from 'react';
import {shallow} from 'enzyme';
import App from './App';

describe('renders without crashing', () => {
  test('if renders', () => {
    const result = shallow(<App/>);
    expect(result).toMatchSnapshot();
  });
});
