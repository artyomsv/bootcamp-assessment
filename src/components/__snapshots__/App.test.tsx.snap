// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing if renders 1`] = `
<App
  classes={
    Object {
      "root": "App-root-1",
    }
  }
/>
`;
